src/args.rs

```diff
@@ -0,0 +1,154 @@
+/// rperf, validates network throughput capacity and reliability,
+/// https://github.com/opensource-3d-p/rperf
+#[derive(clap::Parser, Debug, Clone)]
+#[command(author, version, about, long_about = None)]
+pub struct Args {
+    /// the port used for client-server interactions
+    #[arg(short, long, value_name = "number", default_value_t = 5199)]
+    pub port: u16,
+
+    /// specify logical CPUs, delimited by commas, across which to round-robin affinity;
+    /// not supported on all systems
+    #[arg(short = 'A', long, value_name = "numbers", default_value = "")]
+    pub affinity: String,
+
+    /// bind to the interface associated with the address <host>
+    #[arg(
+        short = 'B',
+        long,
+        conflicts_with = "client",
+        default_value_if("version6", "true", Some("::")),
+        default_value = "0.0.0.0",
+        value_name = "host"
+    )]
+    pub bind: std::net::IpAddr,
+
+    /// emit debug-level logging on stderr; default is info and above
+    #[arg(short, long)]
+    pub debug: bool,
+
+    /// run in server mode
+    #[arg(short, long, conflicts_with = "client")]
+    pub server: bool,
+
+    /// enable IPv6 on the server (on most hosts, this will allow both IPv4 and IPv6,
+    /// but it might limit to just IPv6 on some)
+    #[arg(short = '6', long)]
+    pub version6: bool,
+
+    /// limit the number of concurrent clients that can be processed by a server;
+    /// any over this count will be immediately disconnected
+    #[arg(long, value_name = "number", default_value = "0")]
+    pub client_limit: usize,
+
+    /// run in client mode; value is the server's address
+    #[arg(short, long, value_name = "host", conflicts_with = "server")]
+    pub client: Option<std::net::IpAddr>,
```

Comment:
I get what you're going for here, but this breaks compatibility with hostname-based resolution.

Not everything will, or should, be specified as an IP address.

Reply:
A string?

Reply:
Yeah, specifically a hostname as defined in https://datatracker.ietf.org/doc/html/rfc952 or an IP address, v4 or v6, per https://datatracker.ietf.org/doc/html/rfc1123#page-13, but I'm perfectly content for it to just be a string and for the user to figure it out if they enter something nonsensical, rather than having the tool try to hold their hand on something basic like this.